core/storage/btree.rs

```diff
@@ -3068,15 +3074,37 @@ impl BTreeCursor {
                             // TODO(pere): make this reference and not copy
                             cell_array.cell_payloads.push(to_static_buf(cell_buf));
                         }
-                        // Insert overflow cells into correct place
+                        // Insert overflow cells into correct place.
+                        // Multiple overflow cells may have the same index, so we need to
```

Comment:
They may not have the same index. In fact the only scenario where there are multiple overflow cells is when the insertion of divider cells to the parent _during a balance_ causes the parent to overflow. Even in that case, the divider cells will be in ascending sorted order by index, which means `Vec::insert()` will not cause incorrect shifting.

The above looks like an invariant we should enforce with an assertion, actually.